### DISCUSSION TITLE
Editing hardening model in FEPX

### DISCUSSION THREAD

**Discussion by HitarthM1729** [neperfepx/FEPX#122] [Q&A] [answered]
I have already tried different tutorial examples given in fepx documentation. However, I want to modify the default hardening law equations. Could you please help me which files or folders do I need to look into.

**Comment by mkasemer** [comment, marked as answer]
@HitarthM1729 What hardening laws are you interested in? I share a common goal in wanting to implement different hardening rules, and would be happy to help guide development.

In short, much of the hardening is mostly handled in `/src/libfepx/solveit_evp/solve_evp/solve_evp_rstar/update_hardening/*`. You can review the routines there and let me know what questions you have.

**Reply by HitarthM1729** [reply]
Thank you very much. I am looking to implement kinematic hardening to simulate cyclic loading. I am very new to FEPX. Any help would be appreciated

**Reply by mkasemer** [reply]
Do you mean that you would like to introduce a "backstress" into the model? To account for Bauschinger's effect?

**Reply by HitarthM1729** [reply]
Yes. Exactly. I tried the default cyclic hardening model available in fepx. But I have gone through literature which suggest different models for modeling kinematic hardening. So i wanted to incorporate that in fepx.

**Reply by mkasemer** [reply]
What model, in particular, are you interested in implementing?

**Reply by HitarthM1729** [reply]
![image](https://github.com/user-attachments/assets/ee19b7a2-3a40-4912-a3e9-2fe19cf6c651)
I am interested in modeling this kinematic hardening equation. You can find the link to the paper here:https://doi.org/10.1016/j.ijmecsci.2020.106068
